src/containers/Application.js

```diff
@@ -312,9 +317,36 @@ class Application extends Component {
 			return Promise.reject(error);
 		});
 
-		// We want to remove networking indicator when we recieve the response
+		// We want to remove the networking indicator when we receive the response
 		axios.interceptors.response.use(function (response) {
+			// Call the `popNetworkingIndicator` method to remove the networking indicator (e.g., loading spinner)
 			that.popNetworkingIndicator();
+
+			// Check if the response content type is 'application/json' or starts with 'application/json'
+			if ((response.headers['content-type'] == 'application/json') || (response.headers['content-type']?.startsWith('application/json;'))) {
+				// If the response is JSON, then parse it with respect to BigInt
+				// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse
+				try {
+					// Parse the response data using a custom reviver function
+					response.data = JSON.parse(
+						response.data,
+						// Custom reviver function to handle BigInt values
+						(key, value, context) => {
+							// Check if the key is in the `JSONParseBigInt` set and the value is a number
+							if (that.JSONParseBigInt.has(key) && (typeof value === 'number')) {
+								// Convert the number to a BigInt
+								return BigInt(context.source);
+							}
+							// Return the original value if no conversion is needed
+							return value;
+						}
+					);
+				} catch (e) {
+					// Log any errors that occur during JSON parsing
+					console.error("Error in axios.interceptors.response", e);
+				}
+			}
+			// Return the modified response object
```

Comment:
@aringocode Not sure about this comment - when the response headers will not meet the conditional criteria, the response will be returned in the original form without any modification. If it will meet the conditional criteria, then it may be modified, but only in case there is a bigInt.